entrypoint.sh

```diff
@@ -6,33 +6,45 @@ export HOME="/github/workspace"
 export NVM_DIR="/github/workspace/nvm"
 export WRANGLER_HOME="/github/workspace"
 
-# h/t https://github.com/elgohr/Publish-Docker-Github-Action
-sanitize() {
-  if [ -z "${1}" ]
-  then
-    >&2 echo "Unable to find ${2}. Did you add a GitHub secret called key ${2}, and pass in secrets.${2} in your workflow?"
-    exit 1
-  fi
-}
-
 curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.0/install.sh | bash
+
+# shellcheck source=/dev/null
 [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+# shellcheck source=/dev/null
 [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"
 
 mkdir -p "$HOME/.wrangler"
 chmod -R 770 "$HOME/.wrangler"
 
-sanitize "${INPUT_EMAIL}" "email"
-sanitize "${INPUT_APIKEY}" "apiKey"
+if [ -n "$INPUT_APITOKEN" ]
+then
+  echo "Using API token authentication"
+  export CF_API_TOKEN="$INPUT_APITOKEN"
+fi
+
+if [ -n "$INPUT_APIKEY" ] || [ -n "$INPUT_EMAIL" ]
+then
+  echo "Using legacy email and API key authentication"
+  export CF_EMAIL="$INPUT_EMAIL"
+  export CF_API_KEY="$INPUT_APIKEY"
+fi
 
-export CF_EMAIL="$INPUT_EMAIL"
-export CF_API_KEY="$INPUT_APIKEY"
+if [ -z "$INPUT_APITOKEN" ] && [ -z "$INPUT_EMAIL" ] && [ -z "$INPUT_APIKEY" ]
```

Comment:
this doesn't fail if only one of (email, key) is given, which is insufficient

Reply:
nice catch! I rewrote this a bit to bail out if either api key or email are set, but not both. our CI is a little busted rn so you can see it working in action here: https://github.com/signalnerve/static-test/commit/e85ebdc37f6a9ea47d5c7dc1e191b74b4e007a3f/checks?check_suite_id=349537573